django/db/models/sql/query.py

```diff
@@ -1186,7 +1186,10 @@ def build_filter(self, filter_expr, branch_negated=False, current_negated=False,
 
         if field.is_relation:
             # No support for transforms for relational fields
-            assert len(lookups) == 1
+            if len(lookups) > 1:
```

Comment:
You could add that as an assertion (that is, assert len(lookups) > 0). I think that would make sense - len(lookups) == 0 is an error somewhere in Django's code, that is, it is a condition that really shouldn't ever happen.

Anothre thing we could do here is supply the allowed field names, like we do in other cases where lookup fails. But this can (and probably should) wait for another PR.